File: app/core/src/main/java/com/fsck/k9/setup/ServerNameSuggester.java
package com.fsck.k9.setup;


import com.fsck.k9.mail.ServerSettings.Type;


public class ServerNameSuggester {
    public String suggestServerName(Type serverType, String domainPart) {
        switch (serverType) {
            case IMAP: {
                return "imap." + domainPart;
            }
            case SMTP: {
                return "smtp." + domainPart;
            }
            case WebDAV: {
                return "exchange." + domainPart;
            }
            case POP3: {
                return "pop3." + domainPart;
            }
        }

        throw new AssertionError("Missed case: " + serverType);
    }
}

File: app/core/src/main/java/com/fsck/k9/setup/ServerNameSuggester.kt
package com.fsck.k9.setup

import com.fsck.k9.preferences.Protocols

class ServerNameSuggester {
    fun suggestServerName(serverType: String, domainPart: String): String = when (serverType) {
        Protocols.IMAP -> "imap.$domainPart"
        Protocols.SMTP -> "smtp.$domainPart"
        Protocols.WEBDAV -> "exchange.$domainPart"
        Protocols.POP3 -> "pop3.$domainPart"
        else -> throw AssertionError("Missed case: $serverType")
    }
}
